connectors/job_cleanup.py

```diff
@@ -0,0 +1,58 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+"""
+A task periodically clean up orphaned and stuck jobs.
+"""
+import asyncio
+from connectors.logger import logger
+
+
+class JobCleanUp:
+    def __init__(
+        self,
+        connectors,
+        native_service_types=[],
+        connector_ids=[],
+        interval=300,
+        stuck_threshold=60,
+    ):
+        self.connectors = connectors
+        self.native_service_types = native_service_types.copy()
+        self.connector_ids = connector_ids
+        self.interval = interval
+        self.stuck_threshold = stuck_threshold
+        self.running = False
+        self.task = None
+
+    def start(self):
```

Comment:
to make you life easer, I would remove the sync layer here, and 

- make `start` a coroutine and replace `_cleanup` 
- make `shutdown` a coroutine
- add a `stop` sync method


And users can start it with:

```
cleaner = JobCleanUp()
asyncio.create_task(cleaner.start())
```

and stop it with  
```
cleaner.stop()
```

and in the cleanups, do

```
await cleaner.shutdown()
```